src/app/elements/parking-map.tsx

```diff
@@ -17,32 +18,23 @@ export function ParkingMap() {
 
   const libraries = useMemo(() => ['places'], [])
 
-
   const { isLoaded } = useLoadScript({
     googleMapsApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY as string,
     libraries: libraries as any, // eslint-disable-line @typescript-eslint/no-explicit-any
   })
 
+
   const selectClosestParkingSpot = useCallback((userLatLong: google.maps.LatLng) => {
     const parkingLocations = getAugmentedParkingLocationData();
-    let firstPoint = parkingLocations[0].path[0];
-    let minDistance = google.maps.geometry.spherical.computeDistanceBetween(userLatLong, new google.maps.LatLng(firstPoint.lat, firstPoint.lng));
-    let closestSpot: AugmentedParkingLocationData | null = parkingLocations[0];
-
-    for (const location of parkingLocations) {
-      for (const point of location.path) {
-        const locationLatLong = new google.maps.LatLng(point.lat, point.lng);
-        const distance = google.maps.geometry.spherical.computeDistanceBetween(userLatLong, locationLatLong);
-        if (distance < minDistance) {
-          closestSpot = location;
-          minDistance = distance;
-        }
-      }
-    }
-
+    const userGeohashAndNeighbors = getUserGeohashAndNeighbors(userLatLong, geohashPrecision);
```

Comment:
This might not be necessary, but I was concerned that there was an edge case that we would filter out neighbouring grids, so I included them

Reply:
I think this makes sense, since it would consider spots that aren't in the same exact geohashed region as the user, but might be right across the boundary (even if this is just a matter of meters!). 

Visual depiction:
`[    <SpotFar>                                      <User>][<SpotClose>                                              ]`

For a super performant use case, maybe the user's location within the region can be detected to only use some neighbors - but this won't matter now.

Reply:
Ideally, I think one would use the geohash to fetch the spots from the database in the first case, to avoid loading them all into memory. If so, this filtering is essentially pre-computed, and all you need to figure out is the neighboring geohashes and fetch locations that match.

Reply:
Yes I think thats where I came across it previously, was only fetching locations close the user from db! Thanks for input.